### PR TITLE
Resolves #2696, added the vocation base id and the functions to manage it instead of the "client id"

### LINKED ISSUE #266: [Bug]: Vocation
### What happened?

https://www.youtube.com/watch?v=6Qn-DVRfdus

### How to Reproduce?

1. Log in
2. Press CTRL+G or log out

### Relevant log output

_No response_

### Attach any relevant file

_No response_

### Version

12.x (Default)

### What OS are you seeing the problem on?

Windows

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #2696: [Bug]: Vocation
### What happened?

https://www.youtube.com/watch?v=6Qn-DVRfdus

### How to Reproduce?

1. Log in
2. Press CTRL+G or log out

### Relevant log output

_No response_

### Attach any relevant file

_No response_

### Version

12.x (Default)

### What OS are you seeing the problem on?

Windows

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by beats-dh**
Fix #2696 
New lua function: `vocation:getBaseId()`
New c++ function `vocation->getBaseId()`

The idea of the base id is that the "promotions" have the same id as the base vocation, so that it is not necessary to have repetitions of ifs and if a new vocation is added, it will also not be necessary to register them in the scripts that call the vocations , because the id is the same. This function should only be used in scripts that will be used by all base vocations.

For example, if a spell can be buy by the "knight" or "elite knight", then you can use vocation:getBaseId()
It is always preferable to use vocation:getBaseId() when possible.

### CODE REVIEW

File: data/spells/scripts/monster/prince drazzak sorcerer.lua [hunk 1]
```diff
@@ -45,7 +45,7 @@ function onTargetTile(creature, pos)
 				local player = Player(creatureTable[r])
 
 				if isPlayer(creatureTable[r]) == true
-				and table.contains({VOCATION.CLIENT_ID.CLIENT_ID.SORCERER}, player:getVocation():getClientId()) then
```
Comment (beats-dh) [inline data/spells/scripts/monster/prince drazzak sorcerer.lua:48]: Aqui estava bugado?

Comment (dex-89) [reply data/spells/scripts/monster/prince drazzak sorcerer.lua:48]: Sim, encontrei-o ontem ao verificar os arquivos.

`{VOCATION.CLIENT_ID.CLIENT_ID.SORCERER}` deve ser mudado `{VOCATION.CLIENT_ID.SORCERER}`

Comment (beats-dh) [reply data/spells/scripts/monster/prince drazzak sorcerer.lua:48]: Arrumei já nesse commit.  Vou arrumar no PR das spells também

**Review by dex-89 — APPROVED** [review]

**Comment by dex-89** [comment]
Works perfectly

**Comment by majestyotbr** [comment]
Tested and working.

**Review by majestyotbr — APPROVED** [review]

**Review by dudantas — APPROVED** [review]

**Comment by dudantas** [comment]
Thanks for your contribution!

**Pull request merged**